crates/rolldown_plugin_isolated_declaration/src/lib.rs

```diff
@@ -0,0 +1,78 @@
+use std::{borrow::Cow, path::Path};
+
+use oxc::{
+  allocator::IntoIn,
+  ast_visit::VisitMut,
+  codegen::{CodeGenerator, CodegenOptions},
+  isolated_declarations::{IsolatedDeclarations, IsolatedDeclarationsOptions},
+};
+use rolldown_common::ModuleType;
+use rolldown_plugin::{Plugin, PluginHookMeta, PluginOrder};
+use sugar_path::SugarPath;
+use type_import_visitor::TypeImportVisitor;
+
+mod type_import_visitor;
+
+#[derive(Debug)]
+pub struct IsolatedDeclarationPlugin {}
+
+impl Plugin for IsolatedDeclarationPlugin {
+  fn name(&self) -> Cow<'static, str> {
+    Cow::Borrowed("builtin:isolated-declaration")
+  }
+
+  async fn transform_ast(
+    &self,
+    ctx: &rolldown_plugin::PluginContext,
+    mut args: rolldown_plugin::HookTransformAstArgs<'_>,
+  ) -> rolldown_plugin::HookTransformAstReturn {
+    if matches!(args.module_type, ModuleType::Ts | ModuleType::Tsx) {
+      let type_import_specifiers = args.ast.program.with_mut(|fields| {
+        let mut visitor = TypeImportVisitor { imported: vec![].into_in(fields.allocator) };
+        visitor.visit_program(fields.program);
+        visitor.imported
+      });
+
+      let importer_path = args.cwd.join(args.id);
+      let importer = importer_path.to_string_lossy();
+      for specifier in type_import_specifiers {
+        // TODO load typing importers module, because it will not load after strip typing.
+        let _ = ctx.resolve(&specifier, Some(&importer), None).await??;
+        // ctx.load(&resolved_id.id, None, load_callback_fn)
+      }
+
+      let ret = args.ast.program.with_mut(|fields| {
+        IsolatedDeclarations::new(fields.allocator, IsolatedDeclarationsOptions::default())
```

Comment:
We should support passing the [stripInternal](https://www.typescriptlang.org/tsconfig/#stripInternal) option to IsolatedDeclarations.